File: java/src/test/java/io/binac/leetcode/ReverseLinkedListIITest.java
package io.binac.leetcode;

import io.binac.leetcode.util.LinkedLists;
import io.binac.leetcode.util.ListNode;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ReverseLinkedListIITest {
    private static final ReverseLinkedListII.Solution1 SOLUTION1 = new ReverseLinkedListII.Solution1();

    private final ListNode HEAD1 = LinkedLists.asLinkedList(1, 2, 3, 4, 5);
    private final int LEFT1 = 2, RIGHT1 = 4;
    private final ListNode EXPECTED1 = LinkedLists.asLinkedList(1, 4, 3, 2, 5);

    private final ListNode HEAD2 = LinkedLists.asLinkedList(5);
    private final int LEFT2 = 1, RIGHT2 = 1;
    private final ListNode EXPECTED2 = LinkedLists.asLinkedList(5);

    @Test
    void test1() {
        assertTrue(LinkedLists.equals(EXPECTED1, SOLUTION1.reverseBetween(HEAD1, LEFT1, RIGHT1)));
        assertTrue(LinkedLists.equals(EXPECTED2, SOLUTION1.reverseBetween(HEAD2, LEFT2, RIGHT2)));
    }
}

File: cpp/tests/ReverseLinkedListIITest.cpp
#include "ReverseLinkedListII.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ReverseLinkedListII, Solution92_1) {
  auto S92_1 = Solution92_1();

  auto Head1 = newLinkedList({1, 2, 3, 4, 5});
  int const Left1 = 2, Right1 = 4;
  auto Expected1 = newLinkedList({1, 4, 3, 2, 5});
  Head1 = S92_1.reverseBetween(Head1, Left1, Right1);
  EXPECT_EQ(*Expected1, *Head1);
  deleteLinkedList(Head1);
  deleteLinkedList(Expected1);

  auto Head2 = newLinkedList({5});
  int const Left2 = 1, Right2 = 1;
  auto Expected2 = newLinkedList({5});
  Head2 = S92_1.reverseBetween(Head2, Left2, Right2);
  EXPECT_EQ(*Expected2, *Head2);
  deleteLinkedList(Head2);
  deleteLinkedList(Expected2);
}